orm/src/main/java/org/hibernate/search/cfg/impl/HibernateCoreIdUniquenessResolver.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.search.cfg.impl;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import org.hibernate.cfg.Configuration;
+import org.hibernate.mapping.PersistentClass;
+import org.hibernate.search.cfg.spi.IdUniquenessResolver;
+
+/**
+ * Implementation extracting entity classes from the configuration
+ * and release the configuration reference for future Garbage Collection.
+ *
+ * @author Emmanuel Bernard &lt;emmanuel@hibernate.org&gt;
+ */
+public class HibernateCoreIdUniquenessResolver implements IdUniquenessResolver {
+	private final Set<Class<?>> entities;
+
+	public HibernateCoreIdUniquenessResolver(Configuration cfg) {
```

Comment:
What happens if a user works with [shared indexes](https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#section-sharing-indexes)? Can we account for that here?